include/core/data_containers.h

```diff
@@ -90,5 +94,91 @@ namespace dem_data_containers
+      particle_point_line_contact_info;
+
+    typedef std::unordered_map<
+      types::particle_index,
+      std::tuple<Particles::ParticleIterator<dim>, Point<dim>, Point<dim>>>
+      particle_line_candidates;
+
+    typedef std::unordered_map<
+      types::particle_index,
+      std::pair<Particles::ParticleIterator<dim>, Point<dim>>>
+      particle_point_candidates;
+
+    typedef std::unordered_map<
+      types::particle_index,
+      std::unordered_map<types::boundary_id, Particles::ParticleIterator<dim>>>
+      particle_floating_wall_candidates;
+
+    typedef std::unordered_map<
+      types::particle_index,
+      std::unordered_map<types::boundary_id,
+                         std::tuple<Particles::ParticleIterator<dim>,
+                                    Tensor<1, dim>,
+                                    Point<dim>,
+                                    types::boundary_id,
+                                    types::global_cell_index>>>
+      particle_wall_candidates;
+
+    typedef std::unordered_map<
+      types::particle_index,
+      std::map<types::boundary_id, particle_wall_contact_info_struct<dim>>>
+      particle_wall_in_contact;
+
+    typedef std::unordered_map<
+      types::particle_index,
+      std::unordered_map<types::particle_index,
+                         particle_particle_contact_info_struct<dim>>>
+      adjacent_particle_pairs;
+
+    typedef std::vector<
+      std::map<typename Triangulation<dim - 1, dim>::active_cell_iterator,
+               std::unordered_map<types::particle_index,
+                                  Particles::ParticleIterator<dim>>,
+               cut_cell_comparison<dim>>>
+      particle_floating_mesh_candidates;
+
+    typedef std::vector<
+      std::map<typename Triangulation<dim - 1, dim>::active_cell_iterator,
+               std::unordered_map<types::particle_index,
+                                  particle_wall_contact_info_struct<dim>>,
+               cut_cell_comparison<dim>>>
+      particle_floating_mesh_in_contact;
+
+    typedef std::unordered_map<types::particle_index,
+                               std::vector<types::particle_index>>
+      particle_particle_candidates;
+
+    typedef std::vector<
+      std::vector<typename Triangulation<dim>::active_cell_iterator>>
+      cells_neighbor_list;
+
+    typedef std::vector<std::vector<
+      std::pair<typename Triangulation<dim>::active_cell_iterator,
+                typename Triangulation<dim - 1, dim>::active_cell_iterator>>>
+      floating_mesh_information;
+
+    typedef std::unordered_map<
+      types::global_cell_index,
+      std::vector<typename Triangulation<dim>::active_cell_iterator>>
+      cells_total_neighbor_list;
+
+    typedef std::map<types::boundary_id, std::pair<Tensor<1, 3>, Point<3>>>
+      boundary_points_and_normal_vectors;
+
+    typedef std::map<unsigned int, std::map<types::boundary_id, Tensor<1, 3>>>
+      vector_on_boundary;
+  };
+
```

Comment:
I think for the struct this make sense, I like the current typedef though

Reply:
I'm not sure I agree with this regarding the typedef. The proposed typedef actually indicate what the containers are used to do instead of what they contain. I rather like this.

Reply:
Ok I understand